test/algorithms/length/length_geo.cpp

```diff
@@ -0,0 +1,144 @@
+    typedef typename bg::promote_floating_point
+    <
+    typename bg::coordinate_type<P>::type
+    >::type rtype;
+
+    typedef bg::srs::spheroid<rtype> stype;
+
+    typedef bg::strategy::distance::andoyer<stype> andoyer_type;
+
+    BOOST_CONCEPT_ASSERT
+        (
+            (bg::concept::PointDistanceStrategy<andoyer_type, P, P>)
+        );
+
+    typedef bg::strategy::distance::thomas<stype> thomas_type;
+
+    BOOST_CONCEPT_ASSERT
+        (
+            (bg::concept::PointDistanceStrategy<thomas_type, P, P>)
+        );
+
+    typedef bg::strategy::distance::vincenty<stype> vincenty_type;
+
+    BOOST_CONCEPT_ASSERT
+        (
+            (bg::concept::PointDistanceStrategy<vincenty_type, P, P>)
+        );
+
+    //typedef typename bg::strategy::distance::services::default_strategy
+    //<
+    //    bg::point_tag, bg::point_tag, P
+    //>::type default_strategy_type;
+
+};
+
+std::vector<std::string > Ls_data_gen()
+{
+   std::string arr[] = {"LINESTRING(0 90,1 80,1 70)",
+                        "LINESTRING(0 90,1 80,1 80,1 80,1 70,1 70)",
+                        "LINESTRING(0 90,1 80,1 79,1 78,1 77,1 76,1 75,1 74,1 73,1 72,1 71,1 70)"};
+   std::vector<std::string> Ls_data (arr, arr + sizeof(arr) / sizeof(arr[0]));
+   return Ls_data;
+}
+
+template <typename P>
+void test_default() //this should use andoyer strategy
+{
+    std::vector<std::string > Ls_data = Ls_data_gen();
+
+    for(size_t i=0; i<2; ++i)
+        test_geometry<bg::model::linestring<P> >(Ls_data[i], 1116814.237 + 1116152.605);
+
+    // Geometries with length zero
+    test_geometry<P>("POINT(0 0)", 0);
+    test_geometry<bg::model::polygon<P> >("POLYGON((0 0,0 1,1 1,1 0,0 0))", 0);
+}
+
+template <typename P, typename N, typename Strategy>
+void test_with_strategy(N exp_length, Strategy strategy)
+{
+    std::vector<std::string > Ls_data = Ls_data_gen();
+
+    for(size_t i=0; i<2; ++i)
+        test_geometry<bg::model::linestring<P> >(Ls_data[i], exp_length, strategy);
+
+    // Geometries with length zero
+    test_geometry<P>("POINT(0 0)", 0, strategy);
+    test_geometry<bg::model::polygon<P> >("POLYGON((0 0,0 1,1 1,1 0,0 0))", 0, strategy);
+}
+
+template <typename P>
+void test_andoyer()
+{
+    typename geo_strategies<P>::andoyer_type andoyer;
+    test_with_strategy<P>(1116814.237 + 1116152.605, andoyer);
+}
+
+template <typename P>
+void test_thomas()
+{
+    typename geo_strategies<P>::thomas_type thomas;
+    test_with_strategy<P>(1116825.795 + 1116158.7417, thomas);
+}
+
+template <typename P>
+void test_vincenty()
+{
+    typename geo_strategies<P>::vincenty_type vincenty;
+    test_with_strategy<P>(1116828.8624 + 1116161.625, vincenty);
+}
+
+template <typename P>
+void test_all()
+{
+    test_default<P>();
+    test_andoyer<P>();
+    test_thomas<P>();
+    test_vincenty<P>();
+}
+
+template <typename P>
+void test_empty_input()
+{
+    test_empty_input(bg::model::linestring<P>());
```

Comment:
Empty MLs test case is now added below this line but GitHub shows this comment anyway.